### DISCUSSION TITLE
How to persist session across cells?

### DISCUSSION THREAD

**Discussion by RollsChris** [cmoog/vscode-sql-notebook#38] [Q&A]
I could really do with this.. can you give me some pointers so I can try it out :)

**Comment by cmoog** [comment]
Sure. I think we'd need to adjust the usage of the core `extension <-> driver` interface that's located [here](https://github.com/cmoog/vscode-sql-notebook/blob/master/src/driver.ts#L9-L12). We'd only want to call `getConnection` once, rather than for each cell execution. That invocation is currently located [here](https://github.com/cmoog/vscode-sql-notebook/blob/990ff6ce57df1b3b20903ecc8bdb3a240cc394c6/src/controller.ts#L55).

**Comment by RollsChris** [comment]
I have had to go back to the previous version i could debug with. Which means we are probably going to diverge to much... thanks for your help and giveing me some ideas.
